cmd/river/rivercli/river_cli_test.go

```diff
@@ -69,20 +76,102 @@ var (
 	testMigrationAll = []rivermigrate.Migration{testMigration01, testMigration02, testMigration03} //nolint:gochecknoglobals
 )
 
+type TestDriverProcurer struct{}
+
+func (p *TestDriverProcurer) ProcurePgxV5(pool *pgxpool.Pool) riverdriver.Driver[pgx.Tx] {
+	return riverpgxv5.New(pool)
+}
+
+// High level integration tests that operate on the Cobra command directly. This
+// isn't always appropriate because there's no way to inject a test transaction.
+func TestBaseCommandSetIntegration(t *testing.T) {
+	t.Parallel()
+
+	type testBundle struct {
+		out *bytes.Buffer
+	}
+
+	setup := func(t *testing.T) (*cobra.Command, *testBundle) {
+		t.Helper()
+
+		cli := NewCLI(&Config{
+			DriverProcurer: &TestDriverProcurer{},
+			Name:           "River",
+		})
+
+		var out bytes.Buffer
+		cli.SetOut(&out)
+
+		return cli.BaseCommandSet(), &testBundle{
+			out: &out,
+		}
+	}
+
+	t.Run("DebugVerboseMutallyExclusive", func(t *testing.T) {
+		t.Parallel()
+
+		cmd, _ := setup(t)
+
+		cmd.SetArgs([]string{"--debug", "--verbose"})
+		require.EqualError(t, cmd.Execute(), `if any flags in the group [debug verbose] are set none of the others can be; [debug verbose] were all set`)
+	})
+
+	t.Run("MigrateDownMissingDatabaseURL", func(t *testing.T) {
+		t.Parallel()
+
+		cmd, _ := setup(t)
+
+		cmd.SetArgs([]string{"migrate-down"})
+		require.EqualError(t, cmd.Execute(), `required flag(s) "database-url" not set`)
+	})
+
+	t.Run("VersionFlag", func(t *testing.T) {
+		t.Parallel()
+
+		cmd, bundle := setup(t)
+
+		cmd.SetArgs([]string{"--version"})
+		require.NoError(t, cmd.Execute())
+
+		buildInfo, _ := debug.ReadBuildInfo()
+
+		require.Equal(t, strings.TrimSpace(fmt.Sprintf(`
+River version (unknown)
+Built with %s
+		`, buildInfo.GoVersion)), strings.TrimSpace(bundle.out.String()))
```

Comment:
I would have used require.Contains to do not link the test and the templating

```suggestion
		require.Contains(t, bundle.out.String(), "River version (unknown)")
		require.Contains(t, bundle.out.String(), "Built with " + buildInfo.GoVersion)
```

Reply:
Use of `Contains` does seem plausible.  I think this is a little more of a personal preference thing though — I kind of like how when you match the entire output your tests are more strict. i.e. If there was any extra output in there accidentally, they'd catch the problem. This comes with the tradeoff of possibly being a little more brittle in case of minor whitespace changes that aren't really that important to know about.

Reply:
OK 👍